ui/app/pages/settings/networks-tab/network-form/network-form.component.js

```diff
@@ -211,16 +212,27 @@ export default class NetworkForm extends PureComponent {
     )
   }
 
-  validateUrl = (url, stateKey) => {
-    const invalidUrlErrorMsg = stateKey === 'rpcUrl' ? 'invalidRPC' : 'invalidBlockExplorerURL'
-
-    if (validUrl.isWebUri(url) || (stateKey === 'blockExplorerUrl' && url === '')) {
+  validateRpcUrl = (url, stateKey) => {
+    const { rpcUrls } = this.props
+    if (validUrl.isWebUri(url) || url === '') {
       this.setErrorTo(stateKey, '')
     } else {
       const appendedRpc = `http://${url}`
       const validWhenAppended = validUrl.isWebUri(appendedRpc) && !url.match(/^https?:\/\/$/)
 
-      this.setErrorTo(stateKey, this.context.t(validWhenAppended ? 'uriErrorMsg' : invalidUrlErrorMsg))
+      if (validWhenAppended) this.setErrorTo(stateKey, this.context.t('urlErrorMsg'))
+    }
+
+    if (rpcUrls.includes(url)) {
+      this.setErrorTo(stateKey, this.context.t('urlExistsErrorMsg'))
+    }
+  }
+
+  validateBlockExplorerUrl = (url, stateKey) => {
+    if (validUrl.isWebUri(url) || url === '') {
+      this.setErrorTo(stateKey, '')
+    } else {
+      this.setErrorTo(stateKey, this.context.t('invalidBlockExplorerURL'))
```

Comment:
This is missing the `uriErrorMsg` case, where the input is valid after the scheme is added.